### ISSUE TITLE
Privately hosted Eco server not publicly listing

### ISSUE THREAD

**Issue by parawcliffe** [StrangeLoopGames/EcoIssues#7067]
Hi, 

So after a few attempts on random public servers I have experienced that they seem to have a lot of downtime after hotfixes and patches. 

I decided that i would host my own from my PC and play the game simultaneously. The server is up and running fine now has been live now for four days. Unfortunately the only people that have been able to join I have had to personally give the IPAddress.  This leads me to believe that the server is not showing on the master server list in game. 

I am running the game and server through steam, with about 9 casual players connected. Everything is working fine so far.

I am wondering if anyone can help me figure how to get more players able to join without the inconvenience of having to invite them all individually. (I have seen one or two similar posts online but none that I saw were resolved.)

Some Server settings i think might be relevant -
DetectNAT - False
Public Server - True
No Password
Require Authentication - True
MaxSlots - -1
Allow Debug Calls - True
 World Dimensions - (144,144)
Size Description - 2.07km2

Thanks.

Edit -  I just glanced at the log and have seen this message for the first time. 

"Failed to post server listing to remote service: unable to connect to the remote server"

**Comment by gabgab01** [comment]
same problem here.
basically, i guess when you host the server on your pc, and it detects that steam is running, it will try to use steams networking and fails horrifically.
tried to host a server myself. one dude was able to join me via steam, but the server never got listed in the normal server browser and no one else could join. not even via steam.
all the settings were right.
also posted a ticket, but devs were unable to help me.

basically, hosting servers on your own machine is currently almost not possible

**Comment by parawcliffe** [comment]
Loaded another back up it crashed right at the end with the following message - 

Unhandled Exception: System.Net.Sockets.SocketException: The requested address is not valid in its context
   at System.Net.Sockets.Socket.DoBind(EndPoint endPointSnapshot, SocketAddress socketAddress)
   at System.Net.Sockets.Socket.Bind(EndPoint localEP)
   at System.Net.Sockets.TcpListener.Start(Int32 backlog)
   at Eco.Plugins.Networking.ListenServer.Setup(String ip, Int32 port)
   at Eco.Plugins.Networking.NetworkManager.Run()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart(Object obj)
[24:47] Server Initialization finished in 19,077.3ms

**Comment by MarjieAVolk** [comment]
Well one question I have is, why did you turn off DetectNAT?  Were you having trouble connecting yourself while it was on?  If not, you might try turning that back on.

Also, we are currently working on a complete overhaul of our server connection system, to use a third party library which should handle more cases than we are currently, so it's possible on the next major version that this will just work for you.  (I can't promise, but it's possible :P)

**Comment by MarjieAVolk** [comment]
As for the exceptions you have shown, I am not really sure what happened, but I think your best bet will be to restore a backup.  Assuming you are running the Eco Server tool, you can find these in '[...]\steamapps\common\Eco Server\Storage\Backups'.  Copy the files up into the Storage folder, replacing the existing files.

**Comment by parawcliffe** [comment]
I was following a guide and it recommended turning NAT off and forwarding ports manually if no one could connect.
after disabling NAT players were able to connect using the IPAddress i gave them

**Comment by MarjieAVolk** [comment]
Oh that's odd 🤔

**Comment by parawcliffe** [comment]
the exceptions above are form two separate backup files, which is fun.

**Comment by parawcliffe** [comment]
I think one of them is may be do with a change in local IPAddress after restarting the system, which i  am currently attempting to resolve.

**Comment by parawcliffe** [comment]
So I was able to fix, the crash regarding the IPAddress by changing the Local IP in "steamapps/common/ecoserver/configs" the others.
However i have no clue about. Also the overriding issue that no players can connect without being invited remains.

**Comment by MarjieAVolk** [comment]
What were the bad and good versions of the ip address setting, just out of curiosity?

**Comment by parawcliffe** [comment]
The IP Address just changed within the accepted range on the router from 192.168.1.45->192.168.1.60

**Comment by bingliCheng** [comment]
You can try to delete some of the MOD you've added before.It's much safer to set up an hour per hour in the server.

**Comment by metachronism** [comment]
Closing this out, as w/ 7.4 we've completely rewritten the master server and networking code that should handle this scenario.

If you have a free moment and can try the staging branch on steam to confirm this is fixed, that would be handy!  thanks
